### PR TITLE
Cannot return null for non-nullable field SelectedCustomizableOptionValue.sort_order and Call to a member function getPriceType() on n

### PR DESCRIPTION
**Pull request by XxXgeoXxX**
### Description (*)
Description https://github.com/magento/graphql-ce/issues/474

### Manual testing scenarios (*)
Generate Customer token
Create an empty cart as Registered Customer from GraphQL Client
Perform mutation from the description

Note: Magento get custom options from frontend like mixed data [id=>string, id=>array]. That's why in graphQl query we catch error for multidimensional data(Example: checkbox).
Also 'sort_order' doesn't exist on values level. 

Example for multiple data:
```
mutation {
    addSimpleProductsToCart(
    input: {
      cart_id: "{{cart_id}}"
      cartItems: {
        data: {
          qty: 1
          sku: "test-simple"
        }
        customizable_options: [
          {
            id: 14
            value: "[13,23]"
          }
          {
            id: 23
            value: "test"
          }
        ]
      }
    }
  )
{
    cart {
      items {
        product {
          sku
        }
        ... on SimpleCartItem {
          customizable_options {
            id
            is_required
            sort_order
          }
        }
      }
    }
  }
}
```
### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/Magento/QuoteGraphQl/Model/Cart/AddSimpleProductToCart.php [hunk 1]
```diff
@@ -15,6 +15,7 @@
 use Magento\Framework\GraphQl\Exception\GraphQlNoSuchEntityException;
 use Magento\Framework\Stdlib\ArrayManager;
 use Magento\Quote\Model\Quote;
+use phpDocumentor\Reflection\Types\Mixed_;
```
Comment (naydav) [inline app/code/Magento/QuoteGraphQl/Model/Cart/AddSimpleProductToCart.php]: Please, remove the redundant dependency

File: app/code/Magento/QuoteGraphQl/Model/Cart/AddSimpleProductToCart.php [hunk 2]
```diff
@@ -135,7 +136,9 @@ private function extractCustomizableOptions(array $cartItemData): array
 
         $customizableOptionsData = [];
         foreach ($customizableOptions as $customizableOption) {
-            $customizableOptionsData[$customizableOption['id']] = $customizableOption['value'];
+            $customizableOptionsData[$customizableOption['id']] = $this->convertCustomOptions(
```
Comment (naydav) [inline app/code/Magento/QuoteGraphQl/Model/Cart/AddSimpleProductToCart.php]: Please, cover functionality with API-functional test

**Review by naydav — CHANGES_REQUESTED** [review]

File: app/code/Magento/QuoteGraphQl/Model/Cart/AddSimpleProductToCart.php (hunk 2 above)
Comment (XxXgeoXxX) [reply app/code/Magento/QuoteGraphQl/Model/Cart/AddSimpleProductToCart.php]: Tests were added

File: app/code/Magento/QuoteGraphQl/Model/Cart/AddSimpleProductToCart.php (hunk 1 above)
Comment (XxXgeoXxX) [reply app/code/Magento/QuoteGraphQl/Model/Cart/AddSimpleProductToCart.php]: Done

File: app/code/Magento/QuoteGraphQl/Model/Cart/AddSimpleProductToCart.php [hunk 3]
```diff
@@ -156,4 +158,17 @@ private function createBuyRequest(float $qty, array $customOptions): DataObject
...
         ]);
     }
+
+    /**
+     * @param string $value
+     * @return string|array
+     */
+    private function convertCustomOptions(string $value)
+    {
+        if (substr($value, 0, 1) === "[" ||
```
Comment (naydav) [inline app/code/Magento/QuoteGraphQl/Model/Cart/AddSimpleProductToCart.php]: Do we need `OR` or `AND` condition?

Comment (XxXgeoXxX) [reply app/code/Magento/QuoteGraphQl/Model/Cart/AddSimpleProductToCart.php]: My mistake. We need AND condition.  PR was updated

**Comment by TomashKhamlai** [comment]
Hi, @XxXgeoXxX. Feel free to write me in Slack if I can be useful for you during work on this issue.

**Comment by TomashKhamlai** [comment]
When I have this product [prod_opt.zip](https://github.com/magento/graphql-ce/files/3103257/prod_opt.zip) and create the empty cart as a Customer and add this product to cart by querying
```graphql
mutation addSimpleProductsToCart(
  $cart_id: String!
  $qty: Float!
  $sku: String!
) {
  addSimpleProductsToCart(
    input: {
      cart_id: $cart_id
      cartItems: {
        customizable_options: [
          {
            id: 1
            value: "[1]"
          }
          {
            id: 2
            value: "4"
          }
          {
            id: 3
            value: "7"
          }
          {
            id: 4
            value: "[10]"
          }
        ]
        data: {
          qty: $qty
          sku: $sku
        }
      }
    }
  ) {
    cart {
      items {
        id
        qty
        product {
          sku
          stock_status
        }
        ... on SimpleCartItem {
          customizable_options {
            id
            is_required
            label
            type
            values {
              id
              label
              price {
                value
              }
              sort_order
            }
          }
        }
      }
    }
  }
}
```
with variables
```json
{
  "cart_id": "4nsy2q2ldOGv2hcJqjnRUhH1Ub7r9TOt",
  "qty": 1,
  "sku": "simple-product-with-options"
}
```

I get this problem:

```graphql
{
  "errors": [
    {
      "debugMessage": "Cannot return null for non-nullable field SelectedCustomizableOptionValue.sort_order.",
      "message": "Internal server error",
      "category": "internal",
      "locations": [
        {
          "line": 23,
          "column": 15
        }
      ],
      "path": [
        "addSimpleProductsToCart",
        "cart",
        "items",
        0,
        "customizable_options",
        0,
        "values",
        0,
        "sort_order"
      ]
    },
    {
      "debugMessage": "Cannot return null for non-nullable field SelectedCustomizableOptionValue.sort_order.",
      "message": "Internal server error",
      "category": "internal",
      "locations": [
        {
          "line": 23,
          "column": 15
        }
      ],
      "path": [
        "addSimpleProductsToCart",
        "cart",
        "items",
        0,
        "customizable_options",
        1,
        "values",
        0,
        "sort_order"
      ]
    },
    {
      "debugMessage": "Cannot return null for non-nullable field SelectedCustomizableOptionValue.sort_order.",
      "message": "Internal server error",
      "category": "internal",
      "locations": [
        {
          "line": 23,
          "column": 15
        }
      ],
      "path": [
        "addSimpleProductsToCart",
        "cart",
        "items",
        0,
        "customizable_options",
        2,
        "values",
        0,
        "sort_order"
      ]
    },
    {
      "debugMessage": "Cannot return null for non-nullable field SelectedCustomizableOptionValue.sort_order.",
      "message": "Internal server error",
      "category": "internal",
      "locations": [
        {
          "line": 23,
          "column": 15
        }
      ],
      "path": [
        "addSimpleProductsToCart",
        "cart",
        "items",
        0,
        "customizable_options",
        3,
        "values",
        0,
        "sort_order"
      ]
    }
  ],
  "data": {
    "addSimpleProductsToCart": {
      "cart": {
        "items": [
          {
            "id": "301",
            "qty": 2,
            "product": {
              "sku": "simple-product-with-options",
              "stock_status": "IN_STOCK"
            },
            "customizable_options": [
              {
                "id": 1,
                "is_required": 1,
                "label": "opt1",
                "type": "checkbox",
                "values": [
                  null
                ]
              },
              {
                "id": 2,
                "is_required": 1,
                "label": "opt2",
                "type": "radio",
                "values": [
                  null
                ]
              },
              {
                "id": 3,
                "is_required": 1,
                "label": "opt3",
                "type": "drop_down",
                "values": [
                  null
                ]
              },
              {
                "id": 4,
                "is_required": 1,
                "label": "opt4",
                "type": "multiple",
                "values": [
                  null
                ]
              }
            ]
          }
        ]
      }
    }
  }
}
```

**Comment by TomashKhamlai** [comment]
@XxXgeoXxX, please apply updates from mainline.

**Comment by naydav** [comment]
@TomashKhamlai I merged mainline, please, continue testing the issue

**Comment by naydav** [comment]
@XxXgeoXxX 
> Also 'sort_order' doesn't exist on values level.

Great point! It was a mistake in the initial scheme.

**Pull request merged**